src/decoding.rs

```diff
@@ -1,31 +1,55 @@
+//! # Decoding
+//!
+//! Use the decode function to decode Crockford base 32-encoded strings. This operation can fail;
+//! if it does, you'll get a reasonably useful error instead of a number.
+//!
+//! ```rust
+//! let x = crockford::decode("4zq");
+//! let y = crockford::decode("4ZQ");
+//!
+//! assert_eq!(5111, x.unwrap());
+//! assert_eq!(5111, y.unwrap());
+//! ```
+
 use error::*;
 use UPPERCASE_ENCODING;
 
 const BASE: u64 = 32;
 
+/// Attempts to decode a Crockford base 32-encoded string into a `u64` value.
```

Comment:
This seems a bit limiting to only support `u64` values. I'd expect it to be able to decode to `Vec<u8>`, possibly with some convenience functionality to be able to decode directly to particular integer types.

This should include an example of its own (see the [EXAMPLE guideline](https://rust-lang-nursery.github.io/api-guidelines/documentation.html#c-example)).

Reply:
That's the other library. :)